### PR TITLE
feat(FilePreview): Possible to see a preview of Image, Video and PDF Files

### LINKED ISSUE #255: Files - Add File Preview
**Task:** In Chore if you uploaded a file or had a file in chat, you could open it and preview the file without leaving the app. We want to add that in as well 

**Screen recording:**

Screenshot 2023-02-06 at 11 54 54 AM

### PR DESCRIPTION
**Pull request by lgmarchi**
### 0. To get thumbnail from PDF files: 
It is necessary to install `pdftoppm` lib in our machines, but I am working on PDF and Video thumbnails to not need commands any more

To install this lib, command on macOS is: 

`brew install poppler`

### 1. Now is possible to see thumbnail in large sizes of video, image and pdf files 

https://user-images.githubusercontent.com/63157656/224209095-77b5fd9f-6671-4dcc-834e-a06964c2dd65.mov

### 2. Possible to change theme in main app, and file preview will change theme as well 

https://user-images.githubusercontent.com/63157656/224208929-c115b982-3657-4a64-bde5-aa30497f900a.mov

### 3. Independent of how many windows we have opened, when we close main app now, other windows will close too

https://user-images.githubusercontent.com/63157656/224208884-24abb9be-53ec-443e-af26-3e15c074006f.mov

- 

### Which issue(s) this PR fixes 🔨

- Resolve #255

### CODE REVIEW

**Comment by phillsatellite** [comment]
Hey Lucas! Just a quick question, for the Video files when we preview I noticed it the video but never actually plays 

https://user-images.githubusercontent.com/93608357/224424603-279766a3-ac42-49ba-a7a8-002bb36646e4.mov

**Comment by sdwoodbury** [comment]
> Hey Lucas! Just a quick question, for the Video files when we preview I noticed it the video but never actually plays

It seems like for a preview, not playing the video would be OK. This code is displaying a thumbnail image in a new window, but it hasn't downloaded the file. Without doing that, playing a video is not possible. 

If we want to play the video, it should be possible. We'd just have to fetch the file from IPFS. The CID is included in the `File` struct.

**Review by sdwoodbury — CHANGES_REQUESTED** [review]
I like the feature. I think the window management needs some changes. 
- If I close the main window, the child window doesn't automatically close. 
- State only tracks one window ID for file previews, but I can open multiple file previews. If I close the main window, Uplink won't know the window IDs of all the windows it needs to close.

**Comment by lgmarchi** [comment]
> > Hey Lucas! Just a quick question, for the Video files when we preview I noticed it the video but never actually plays
> 
> It seems like for a preview, not playing the video would be OK. This code is displaying a thumbnail image in a new window, but it hasn't downloaded the file. Without doing that, playing a video is not possible.
> 
> If we want to play the video, it should be possible. We'd just have to fetch the file from IPFS. The CID is included in the `File` struct.

It could be done in a next PR.

**Comment by sdwoodbury** [comment]
review in progress

File: common/src/state/ui.rs [hunk 1]
```diff
@@ -212,6 +229,24 @@ impl DebugLogger {
...
+        Self {
+            file_preview_window_id,
+        }
+    }
+
+    pub fn take_window_id(&mut self) -> Option<WindowId> {
+        self.file_preview_window_id.take()
+    }
+}
+
```
Comment (sdwoodbury) [inline common/src/state/ui.rs:249]: I forgot to remove this in the PR. It's dead code at this point.

**Review by sdwoodbury — APPROVED** [review]

**Comment by phillsatellite** [comment]
@lgmarchi Hey my friend, so when re-tested I found some things. I think most can probably be fixed in a future PR but the one I was concerned about is the Windows issue

MacOS will close all the previews when I close the app but there will be an error in console (this seems to be different than the error we talked about in chat)

Screenshot 2023-03-14 at 4 05 43 PM

On **Linux** when I closed the app the previews remained open and I had to close them manually 

https://user-images.githubusercontent.com/93608357/225126709-89f3c619-641a-4109-8f6f-2cb693ae02c6.mov


On **Windows** the file preview seemed to have broke all together. I tested this on a Windows VM and actual Windows both Windows 11 tested with old and new accounts.

https://user-images.githubusercontent.com/93608357/225126569-6aee8cb8-1d51-462c-ad5c-4c59ceae5973.mov

**Comment by dariusc93** [comment]
> MacOS will close all the previews when I close the app but there will be an error in console (this seems to be different than the error we talked about in chat)
> 
> Screenshot 2023-03-14 at 4 05 43 PM
> 
The error seems related to #441 , which may not be related to this PR but something separately

**Comment by phillsatellite** [comment]
Currently reviewing again, right now on Windows the file preview seems to work again. Going to investigate a little more on  Windows and other OS's

**Review by phillsatellite — APPROVED** [review]

**Comment by phillsatellite** [comment]
Approved on my side the preview worked on all the OS's / VM's I could still replicate the previews persisting after closing the app on Linux and Windows I can create a ticket for that 2nd dev review and good to go 👍

**Comment by lgmarchi** [comment]
> @lgmarchi Hey my friend, so when re-tested I found some things. I think most can probably be fixed in a future PR but the one I was concerned about is the Windows issue
> 
> MacOS will close all the previews when I close the app but there will be an error in console (this seems to be different than the error we talked about in chat)
> 
> Screenshot 2023-03-14 at 4 05 43 PM
> 
> On **Linux** when I closed the app the previews remained open and I had to close them manually
> 
> On **Windows** the file preview seemed to have broke all together. I tested this on a Windows VM and actual Windows both Windows 11 tested with old and new accounts.

It will be better to treat those problems in other PR, specific for each platform, thanks for all tests my friend!

**Pull request merged**